sympy/ntheory/generate.py

```diff
@@ -142,22 +142,30 @@ def extend_to_no(self, i):
         while len(self._list) < i:
             self.extend(int(self._list[-1] * 1.5))
 
-    def primerange(self, a, b):
-        """Generate all prime numbers in the range [a, b).
+    def primerange(self, a, b=None):
+        """Generate all prime numbers in the range [a, b) when both a and b are
+           provided and generates all prime numbers till a when only a is provided.
 
         Examples
         ========
 
         >>> from sympy import sieve
         >>> print([i for i in sieve.primerange(7, 18)])
         [7, 11, 13, 17]
+
+        >>> print([i for i in sieve.primerange(18)])
+        [2, 3, 5, 7, 11, 13, 17]
```

Comment:
Maybe it would be better to switch the examples to use a prime number as the endpoint, to make it clearer that the endpoint is not included (this can also be explicitly noted in a comment in the text). For example

```
        >>> # Note the endpoint 17 is not included even though it is prime
        >>> print([i for i in sieve.primerange(7, 17)])
        [7, 11, 13]
```